misc/package/Data Files/MWSE/core/mwse/traveled/main.lua

```diff
@@ -0,0 +1,29 @@
+local timeBeforeTravel = 0
+local payload = {}
+
+---@param e calcTravelPriceEventData
+local function travelStart(e)
+	if (not tes3.mobilePlayer.traveling) then -- Get time before traveling
+		timeBeforeTravel = tes3.getSimulationTimestamp()
+
+	else -- Travel finished, but calcTravelPrice triggered once again.
+		table.copy(e, payload)
+		payload.claim = nil
+		payload.hoursPassed = tes3.getSimulationTimestamp() - timeBeforeTravel
+		payload.previousCell = tes3.mobilePlayer.cell
+		timeBeforeTravel = 0
+
+		-- calcTravelPrice event is triggered once for every possible destination on
+		-- first opening of the Travel Menu. Once again after a destination has been
+		-- selected, but before the player was teleported. This event is triggered
+		-- after the player is teleported to the destination.
+		timer.frame.delayOneFrame(function()
+			payload.cell = tes3.mobilePlayer.cell
+			event.trigger(tes3.event.traveled, payload)
+		end)
```

Comment:
Is there something I'd have to do to make this event filterable? Just pass the third table argument with the filter field? Should I put `payload.cell` as the filter?

Reply:
Yes, you could just append `{filter = element.name}` as a third param to trigger. Cell seems reasonable to me.

Reply:
Also, if possible, never use delayOneFrame if possible at all. Those aren't exactly reliable and there might often be a better way to implement it. Registering a doOnce simulate there wont properly catch it?

Reply:
Simulate registered with doOnce works like a charm. Also, I added payload.cell as the filter.